src/pages/ReimbursementAccount/EnableStep.js

```diff
@@ -1,56 +1,131 @@
+import _ from 'underscore';
 import React from 'react';
-import {Image, View} from 'react-native';
+import PropTypes from 'prop-types';
+import {View, Image} from 'react-native';
+import {withOnyx} from 'react-native-onyx';
 import styles from '../../styles/styles';
 import withLocalize, {withLocalizePropTypes} from '../../components/withLocalize';
-import {navigateToConciergeChat} from '../../libs/actions/Report';
 import HeaderWithCloseButton from '../../components/HeaderWithCloseButton';
 import Navigation from '../../libs/Navigation/Navigation';
 import Text from '../../components/Text';
-import CONST from '../../CONST';
-import TextLink from '../../components/TextLink';
 import compose from '../../libs/compose';
+import ONYXKEYS from '../../ONYXKEYS';
+import {ChatBubble} from '../../components/Icon/Expensicons';
+import MenuItem from '../../components/MenuItem';
+import getBankIcon from '../../components/Icon/BankIcons';
+import {getPaymentMethods} from '../../libs/actions/PaymentMethods';
+import FullScreenLoadingIndicator from '../../components/FullscreenLoadingIndicator';
+import bankAccountPropTypes from '../../components/bankAccountPropTypes';
+import {navigateToConciergeChat} from '../../libs/actions/Report';
+import confettiPop from '../../../assets/images/confetti-pop.gif';
+import Icon from '../../components/Icon';
+import WorkspaceSection from '../workspace/WorkspaceSection';
+import {ConciergeBlue} from '../../components/Icon/Illustrations';
 
 const propTypes = {
+    /** Are we loading payment methods? */
+    isLoadingPaymentMethods: PropTypes.bool,
+
+    /** Array of bank account objects */
+    bankAccountList: PropTypes.arrayOf(bankAccountPropTypes),
+
     ...withLocalizePropTypes,
 };
 
-const EnableStep = ({translate}) => {
-    const verifyingUrl = `${CONST.CLOUDFRONT_URL}/images/icons/emptystates/emptystate_reviewing.gif`;
-    return (
-        <View style={[styles.flex1, styles.justifyContentBetween]}>
-            <HeaderWithCloseButton
-                title={translate('validationStep.headerTitle')}
-                onCloseButtonPress={Navigation.dismissModal}
-            />
-            <View style={[styles.flex1]}>
-                <Image
-                    source={{uri: verifyingUrl}}
-                    style={[styles.workspaceInviteWelcome]}
-                    resizeMode="center"
+const defaultProps = {
+    isLoadingPaymentMethods: true,
+    bankAccountList: [],
+};
+
+class EnableStep extends React.Component {
+    componentDidMount() {
+        getPaymentMethods();
+    }
+
+    render() {
+        const {
+            user, reimbursementAccount, translate, bankAccountList,
+        } = this.props;
+        if (this.props.isLoadingPaymentMethods || _.isEmpty(bankAccountList)) {
+            return (
+                <FullScreenLoadingIndicator visible />
```

Comment:
I assume that this step isn't going to load if the app goes offline? I ran into this in a few places, and I ended up just removing the loaders and letting the view render with whatever data was already in Onyx while the new data is loading (and then when the app comes online, the request finishes and the new data loads into Onyx).

Reply:
> I don't see why this page wouldn't show if you were offline.

I'm thinking about a case where you've got the app open, you go offline, and then navigate to this step. The API call is triggered for fetching the paymentMethods and the loading screen comes up. However, since you're offline, the loading screen will never go away (until you come back online).

My sense is that generally we wouldn't want the user to just be there looking at a loading indicator if we can simply show them whatever is already in Onyx. 

I realize that can be on a case-by-case basis though, so it might not work in this case.

Reply:
>I realize that can be on a case-by-case basis though, so it might not work in this case.

Yeah I think we will just need to investigate this and re-visit how that would work. 

It's debatable, but I think there's a case for not letting a user try to add a bank account at all if they are offline and don't yet have one added. The entire experience will be subpar because we need to validate their input in the backend at each step, load the Plaid SDK, etc. People just won't be able to get very far in the mainline flow.

That said, I can definitely see the case where local storage already knows you are "set up" and we should expect to see the "All done!" or "One more thing!" screens.